pilot/pkg/model/sidecar.go

```diff
@@ -216,25 +216,55 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *Config, configNamespa
 
 	// Assign namespace dependencies
 	out.namespaceDependencies = make(map[string]struct{})
+
+	addService := func (s *Service) {
+		if s == nil {
+			return
+		}
+		if foundSvc, found := servicesAdded[string(s.Hostname)]; !found {
+			servicesAdded[string(s.Hostname)] = s
+			out.services = append(out.services, s)
+			out.namespaceDependencies[s.Attributes.Namespace] = struct{}{}
+		} else if foundSvc.Attributes.Namespace == s.Attributes.Namespace && s.Ports != nil && len(s.Ports) > 0 {
+			// merge the ports to service when each listener generates partial service
+			// we only merge if the found service is in the same namespace as the one we're trying to add
```

Comment:
I'd like to call to attention this new requirement; we might attempt to add an inferred service with the same hostname but a different namespace to one specifically required in the hosts field, and we should do nothing in that case

Reply:
I don't think that this new restriction will matter for concretely imported services, listener.services is unique in hostnames